core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala

```diff
@@ -1298,19 +1298,17 @@ class SizeBasedCoalescer(val maxSize: Int) extends PartitionCoalescer with Seria
       val splitSize = fileSplit.getLength
       if (currentSum + splitSize < maxSize) {
         addPartition(partition, splitSize)
-        index += 1
-        if (index == partitions.size) {
-          updateGroups
-        }
       } else {
-        if (currentGroup.partitions.size == 0) {
+        if (currentGroup.partitions.isEmpty) {
           addPartition(partition, splitSize)
-          index += 1
         } else {
-          updateGroups
+          updateGroups()
+          addPartition(partition, splitSize)
         }
```

Comment:
Instead of the following,
```scala
if (currentGroup.partitions.isEmpty) {
  addPartition(partition, splitSize)
} else {
  updateGroups()
  addPartition(partition, splitSize)
}
```

The following will be better.
```scala
if (currentGroup.partitions.nonEmpty) {
  updateGroups()
}
addPartition(partition, splitSize)
```

Reply:
Updated, thanks